### ISSUE TITLE
batch_size in model %>% fit

### ISSUE THREAD

**Issue by Xabiermm1** [rstudio/keras#984]
Hi, 

I am not very experienced in NN so maybe I am missunderstanding something but:

batch_size max value should be the sample size no?
For example I have a matrix of 1000x6000 size. The max value for batch_size  should by 1000 no?. So we I can put 100000000 as a valid value for batch_size   and the model trains without any problem?

Xabi

**Comment by henry090** [comment]
Hi @Xabiermm1 , I have just tested it and it works.

### Input
```
x_data <- matrix(data = runif(500,0,1),nrow = 1e3,ncol = 6e3)
label <-  ifelse(runif(1e3,0,1) > 0.6, 1L,0L) %>% as.matrix()

library(dplyr);library(keras)

keras_model_sequential() %>% 
  layer_dense(100) %>% 
  layer_dense(1) %>% 
  compile(
    loss = 'binary_crossentropy',
    optimizer = optimizer_rmsprop(),
    metrics = c('accuracy')
  ) %>% 
  fit(
    x_data,label, 
    epochs = 2, batch_size = 100e6, 
    validation_split = 0.2
  )
```
### Output 

```
Train on 800 samples, validate on 200 samples
Epoch 1/2
800/800 [==============================] - 0s 604us/sample - loss: 4.0172 - acc: 0.4762 - val_loss: 6.3242 - val_acc: 0.5900
Epoch 2/2
800/800 [==============================] - 0s 77us/sample - loss: 6.1700 - acc: 0.6000 - val_loss: 6.3242 - val_acc: 0.5900
```

**Comment by Xabiermm1** [comment]
ok, So base on the example you are showing you have 1000 samples.

If you put inside your

fit(
    x_data,label, 
    epochs = 2, batch_size = 100e6, 
    validation_split = 0.2
    batch_size = 6000,                        <- this will works but it shouldn't? why?
  )

**Comment by dfalbel** [comment]
The following chunk will show that it uses the sample_size if the batch_size is larger than sample_size. See:

``` r
library(keras)
x <- matrix(rep(0, 100), ncol = 1)
model <- keras_model_sequential() %>% 
  layer_dense(1)

closs <- function(ytrue, ypred) {
  tensorflow::tf$print(tensorflow::tf$shape(ypred))
  loss_mean_absolute_error(ytrue, ypred)
}
model %>% compile(loss = closs , optimizer = "adam")

model %>% 
  fit(x, x, batch_size = 150, epochs = 1)
```

<sup>Created on 2020-02-14 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by Xabiermm1** [comment]
OK!! a lot of thank. 
Maybe a error or warning should by interesting in order to notice this to the user, else it's diff to understand why keep working when shouldn't.

**Comment by dfalbel** [comment]
You are right! I would be happy to review a PR if you want to add this warning :)
